website/docs/examples/monitor_app/README.md

```diff
@@ -0,0 +1,385 @@
+useEffect(() => {
+    const fetchData = async () => {
+        try {
+            // Replace with your backend URL
+            const response = await fetch(
+                `${window.location.protocol}//${window.location.hostname}:8002/list_uris`
+            );
+            // Check if the request was successful
+            if (!response.ok) {
+                throw new Error('Network response was not ok ' + response.statusText);
+            }
+            const rawData = await response.json();
+            setUris(Object.keys(rawData));
+            // Select the first URI by default
+            setSelectedUri(Object.keys(rawData)[0]);
+        } catch (error) {
+            console.error('Error fetching data:', error);
+        }
+    };
+
+    fetchData();
+  }, []);
+```
+
+The next step is to implement the event function to subscribe to the events using the
+`WebSocket` API and the `useEffect` hook that will be triggered every time we select
+ a new uri. The result is stored in the `details` variable.
+
+```typescript
+useEffect(() => {
+  if (!selectedUri) return;
+
+  const detailSocket = new WebSocket(
+      `ws://${window.location.hostname}:8002/subscribe/${selectedUri}`
+  );
+
+  detailSocket.onopen = (event) => {
+      console.log('Detail WebSocket connection opened:', event);
+  };
+
+  detailSocket.onmessage = (event) => {
+      const receivedDetails = JSON.parse(event.data);
+      setDetails(receivedDetails);
+  }
+
+  detailSocket.onclose = (event) => {
+      console.log('Detail WebSocket connection closed:', event);
+  };
+
+  return () => {
+      detailSocket.close();
+  };
+}, [selectedUri]);
+```
+
+We finally put all together in the `TopicMonitor` component as follows:
+
+```typescript
+return (
+  <div>
+      <select value={selectedUri} onChange={(e) => setSelectedUri(e.target.value)}>
+          {uris.map((uri, index) =>
+              <option key={index} value={uri}>
+                  {uri}
+              </option>
+          )}
+      </select>
+
+      <div>
+          {selectedUri && (
+              <JsonView data={details} shouldExpandNode={allExpanded} style={defaultStyles} />
+          )}
+      </div>
+  </div>
+);
+```
+
+## Setup
+
+First, we need to install the dependencies of the backend and the frontend.
+
+### 1. Install the [farm-ng Brain ADK package](/docs/brain/brain-install)
+
+### 2. Install the backend dependencies
+
+:::tip
+
+It is recommended to also install these dependencies and run the
+example in the brain ADK virtual environment.
+
+:::
+
+```bash
+# assuming you're already in the amiga-dev-kit/ directory
+cd farm-ng-amiga/py/examples/monitor_app
+```
+
+```bash
+pip3 install -r requirements.txt
+```
+
+### 3. Install the frontend dependencies
+
+Recommendation: Use [**Node Version Manager**](
+  <https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04>
+  #option-3-installing-node-using-the-node-version-manager) to install Node.js 18.x.
```

Comment:
I had some problems here.
When simply doing:
```bash
sudo apt install nodejs
sudo apt install npm
```
Apparently, the node version installed (v10.19.0) is not compatible with the `vite` version.
I had to do:
```bash
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
nvm install --lts
nvm use --lts
```

I am not sure if you had the same problem or if we should worry about adding more steps here, but I thought I'd let you know